### ISSUE TITLE
Autocompletion not working

### ISSUE THREAD

**Issue by vrobles-gee** [daniele77/cli#45]
Hello.

I am facing issues with autocompletion. I noticed it works as expected when using a telnet session, but I tested all the other ones (local, file, async) and everytime I try to autocomplete, only the tab is printed as any other character, and the cli gets messed up.

I thought I was doing something wrong, but I tested the examples (generated following the README.md instructions) and experienced the same behavior. I also tested both latest master and v1.1.0 tags.

I am testing using Ubuntu 18.04. Boost version is 1.68. I observe the same behavior in regular terminal as inside vscode terminal. I this a bug, or I am doing something wrong?

Thanks in advance

**Comment by vrobles-gee** [comment]
In case this helps, history doesn't work either (up and down arrows). It does work in telnet as with autocompletion. When using other kind of session, I only see this output (up/down):

cli> ^[[A
cli> ^[[B

**Comment by daniele77** [comment]
Hi @vrobles-gee ,
thank you very much for using and testing the library.

The autocompletion and the history features are available only in the `complete` and `pluginmanager` examples because they use `CliLocalTerminaSession`.

The examples `simplelocalsession` and `filesession`  use `CliFileSession` instead, that does not provide "special keys" handling.

The feature is available in the example `asyncsession` too, but it can be compiled only on POSIX machines (e.g., Linux) because it uses the boost async IO only available in POSIX.

Bottom line: if you need history and autocompletion in your application, you should instantiate `CliLocalTerminaSession`.

**Comment by vrobles-gee** [comment]
Hi Thanks for the fast response.

Just wanted to let you know that I tried both examples, and they work as expected. 

I also want to leave a comment here, because it may help someone else. When I tried to use CliLocalTerminaSession in my application, it was just hanging or returning immediately. The problem is that I started based on the "complete" example, and removed the telnet server.

Then I checked the "pluginmanager" example, and noticed I was missing the work guard:

`boost::asio::make_work_guard`

Hopefully it will help others that face the same issue, because it was not trivial for me, I guess it is for somebody with more experience using boost::asio.

It is working great now, thanks for your help again.

**Comment by daniele77** [comment]
You're right.
Experienced `boost::asio` users know how to use it, but not everyone does.
Maybe it would be useful to add some information about the examples and the usage of the library. I only have to figure out where to put the extra documentation :-)
